File: C++/binarysearch.cpp
#include<iostream>
using namespace std;
int binarysearch(int arr[], int size, int key)
{
    int start = 0;
    int end = size - 1;
    while (start <= end)
    {
        int mid = (start + end)/2;
        if (arr[mid] < key)
            start = mid + 1;
        else if (arr[mid] > key)
            end = mid - 1;
        else if (arr[mid] == key)
            return mid;
    }
}
int main()
{
    int size; cin >> size;
    int arr[size];
    for (int i = 0; i < size; i++)
    {
        cin >> arr[i];
    }
    cout << "Enter element: ";
    int key; cin >> key; 
    cout << "Index : " << binarysearch(arr, size, key);
    return 0;
}

File: python/binarySearch.py
def binarySearch(arr, size, key):
    left = 0
    right = size - 1
    mid = 0
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] < key:
            left = mid + 1
        elif arr[mid] > key:
            right = mid - 1
        else:
            return mid
    return -1
size = int(input("Enter size of array: "))
arr = []
for i in range(0,size):
    a = int(input())
    arr.append(a)
print(arr)
key = int(input('Enter element to be searched: '))
index = binarySearch(arr, size, key)
if index == -1:
    print('Element not found')
else:
    print('Element found at index:', index)